### PR TITLE
Bug 1194097 - [browser] Firefox Account sync welcome page

### PR DESCRIPTION
**Pull request by begeeben**
- Add firefox_account_ui.js for sync welcome settings page.
- Borrow sync_manager_bridge.js from apps/system.
- Add settings permission for changing sync settings.
- Add IAC connection 'gaia-sync-management' in smart-system.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6a869a92d505cd57c582e99df2217a6e563451a1

File: tv_apps/browser/js/firefox_account.js [hunk 1]
```diff
@@ -57,15 +61,34 @@
...
+      this.syncHistory = settings.syncHistory;
+      this.syncPasswords = settings.syncPasswords;
+      this.syncTab = settings.syncTab;
+
+      var promise = new Promise((resolve, reject) => {
+        var lock = navigator.mozSettings.createLock();
+        var result = lock.set({
+          'sync.collections.bookmarks.enabled': this.syncBookmarks,
+          'sync.collections.history.enabled': this.syncHistory,
+          'sync.collections.passwords.enabled': this.syncPasswords
```
Comment (weilonge) [inline tv_apps/browser/js/firefox_account.js]: In the current TV scope, do we have to add Password sync feature?

Comment (weilonge) [inline tv_apps/browser/js/firefox_account.js]: do this remain for future use?

File: tv_apps/browser/index.html [hunk 2]
```diff
@@ -744,6 +751,44 @@ <h1 id="private-browsing-subtitle-text" data-l10n-id="WB_LT_PRIVATE_BROWSING_1">
...
+            <li class="settings-horizontal-list-item">
+              <input type="checkbox" value="bookmarks" id="fxa-welcome-custom-bookmarks-button" name="fxa-welcome-custom-options" class="settings-list-checkbox" checked disabled>
+              <label for="fxa-welcome-custom-bookmarks-button"></label>
+              <span data-l10n-id="bookmarks"></span>
+            </li>
+            <li class="settings-horizontal-list-item">
+              <input type="checkbox" value="history" id="fxa-welcome-custom-history-button" name="fxa-welcome-custom-options" class="settings-list-checkbox" disabled>
+              <label for="fxa-welcome-custom-history-button"></label>
+              <span data-l10n-id="history"></span>
+            </li>
```
Comment (weilonge) [inline tv_apps/browser/index.html]: Add // ToDo: for password and tab.

**Pull request closed without merging**